hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -728,7 +728,7 @@ def lists(
     will be drawn.
 
     If unique is True (or something that evaluates to True), we compare direct
-    object equality, as if unique_by was ``lambda x: x``. This comparison only
+    object equality, as if unique_by was :obj:`~python:hash`. This comparison only
```

Comment:
This is wrong (I hope!). We don't consider two unequal objects with a hash collision to be the same in this case.

Reply:
Phew!  Yeah, I had this reported in person but having run some stress-tests again it's not happening :man_facepalming: 

Just the guides changes, then.